libnetwork/drivers/bridge/setup_ip_tables_linux.go

```diff
@@ -47,7 +47,17 @@ func setupIPChains(config configuration, version iptables.IPVersion) (natChain *
 
 	natChain, err := iptable.NewChain(DockerChain, iptables.Nat, hairpinMode)
 	if err != nil {
-		return nil, nil, nil, nil, fmt.Errorf("failed to create NAT chain %s: %v", DockerChain, err)
+		if version == iptables.IPv4 {
+			return nil, nil, nil, nil, fmt.Errorf("failed to create NAT chain %s: %v", DockerChain, err)
+		}
```

Comment:
This is policy. It doesn't belong in low-level code. Return an error which could be asserted on using `errors.As` or `errors.Is`, and lift the decision on how to handle failure into the caller `func (*driver) configure`.